### DISCUSSION TITLE
Linking a post to multiple topics

### DISCUSSION THREAD

**Discussion by agraashnb** [austintoddj/canvas#1204] [General]
Is there a way to bind a post to multiple topics. Basically one to many relationship between post and topic.

I want to use "topic" for the actual topics that the post may belong to ( And tags as keywords to be used for filters and more).

Currently, the admin allows only one topic to be assigned. Is there a way to change that?

Thanks for your time

Ash

**Comment by austintoddj** [comment]
Hi @agraashnb 

You're right in calling out that the [current relationship](https://github.com/austintoddj/canvas/blob/master/src/Models/Post.php#L106) is misleading. The next version (7) of Canvas addresses that issue.

The separation of tags and topics is intentional, and I don't plan on giving those entities the same functionality. 

I apologize if it's not the answer you're looking for, but let me know if you've got more use cases or features you're interested in.

**Comment by agraashnb** [comment]
Awesome. Thanks for looking into it.

Are you planning to release v7 soon, or is something that would take a lot of time. 

Thanks again
Ash

**Comment by austintoddj** [comment]
I don't have a launch date in mind at the moment other than in 2023.

**Comment by tvarga94** [comment]
@austintoddj any update on that? If I am not mistaken we are still waiting for version 7, but would be great to assign than 1 topic to a post. Any workaround to do that?

**Reply by austintoddj** [reply]
There isn't a workaround right now for assigning multiple topics to a post.

I still haven't been able to dedicate the time needed to finish v7.0. Life has taken some new and unexpected turns, and my free time goes into owning and managing [texastwistersgym.com](https://texastwistersgym.com/). It's the first business I've ever started, so it's pretty exciting 🙂 

I'm sure that's not the answer you were looking for, but I still intend to maintain the project and keep it compatible with future versions of Laravel.

**Comment by austintoddj** [comment]
Closing this out with [Canvas v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0).

v7 models taxonomy as **one topic per post** (`topic_id` FK) plus many tags. Multi-topic pivots are not supported and are not coming back in this major.

Use **tags** for multi-label filtering; keep **topic** as the primary category.
